x-pack/examples/triggers_actions_ui_example/public/connector_types/system_log_example/system_log_example_params.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import type { ActionParamsProps } from '@kbn/triggers-actions-ui-plugin/public';
+import { TextAreaWithMessageVariables } from '@kbn/triggers-actions-ui-plugin/public';
+import { SystemLogActionParams } from '../types';
+
+export const ServerLogParamsFields: React.FunctionComponent<
+  ActionParamsProps<SystemLogActionParams>
+> = ({
+  actionParams,
+  editAction,
+  index,
+  errors,
+  messageVariables,
+  defaultMessage,
+  useDefaultMessage,
+}) => {
+  const { message } = actionParams;
+
+  const [[isUsingDefault, defaultMessageUsed], setDefaultMessageUsage] = useState<
+    [boolean, string | undefined]
+  >([false, defaultMessage]);
+  useEffect(() => {
+    if (
+      useDefaultMessage ||
+      !actionParams?.message ||
+      (isUsingDefault &&
+        actionParams?.message === defaultMessageUsed &&
+        defaultMessageUsed !== defaultMessage)
+    ) {
```

Comment:
should we have unit test and/or explanation for Future Engineer?

Reply:
This is copy-pasted from Server Log, so I'll add a note to refer to that